nilearn/maskers/nifti_labels_masker.py

```diff
@@ -242,6 +245,17 @@ def __init__(
                 f"parameter: {resampling_target}"
             )
 
+        # if not isinstance(labels, (list, np.ndarray)):
+        #     raise TypeError(
+        #         f"'labels' must be a list. Got: {type(labels)}"
+        #     )
+        # if not all(isinstance(x, (str, np.bytes_)) for x in labels):
+        #     raise TypeError(
+        #         "All elements of 'labels' must be a string.\n"
+        #         f"Got a list of {set([type(x) for x in labels])}"
+        #     )
```

Comment:
these checks should probably be implemented especially for the reporting section